codalab/apps/web/templates/web/competitions/view.html

```diff
@@ -229,52 +229,88 @@ <h3 class="panel-title"><strong>Competition Ends</strong></h3>
         </div>
     </section>
         <div class="row">
-        <div class="col-md-4">
-            {% if top_three_leaders and not active_phase.is_blind %}
-                <div class="panel">
-                    <div class="panel-body">
-                        <center><h4>Top Three</h4></center>
-                        <table class="table table-responsive">
-                            <thead>
-                                <tr>
-                                    <th>Rank</th>
-                                    <th>Username</th>
-                                    <th>Score</th>
-                                </tr>
-                            </thead>
-                            <tbody>
-                                {% for leader in top_three_leaders %}
-                                    <tr>
-                                        <td class="leader">{{ forloop.counter }}</td>
-                                        <td class="leader">{{ leader.username }}</td>
-                                        <td class="leader">{{ leader.score }}</td>
-                                    </tr>
+        <div class="col-md-4 well">
+            {% for group in top_three %}
+                <table class="table">
+                    <tr class="">
+                        <th rowspan="2">#</th>
+                        <th rowspan="2">User</th>
+                        {% if phase.competition.allow_teams %}
+                            <th rowspan="2">Team Name</th>
+                        {% endif %}
+                        {% for head in group.headers|slice:"0:1"%}
```

Comment:
Could you do these operations in Python and just pass the same thing as before to top 3?